Migrate system-builder to Ruby 2.5

duduribeiro opened the pull request:
This PR migrates our system-builder image to use Ruby 2.5

gsaslis committed:
Ruby 2.5 Software Collections base image

duduribeiro committed:
Install file command from yum

gsaslis commented on Dockerfile:
```diff
@@ -30,6 +30,7 @@ RUN yum install -y git \
                    make \
                    sudo \
                    rh-nodejs8 \
+                   file \
```
@duduribeiro not that it's a problem or anything, just wondering why this is needed?

duduribeiro replied:
@gsaslis we need it since ImageMagick uses it to verify if a file is an image or not.

our ruby-24-centos image had it, but it was removed from ruby-25-centos.

gsaslis reviewed (approved):
lgtm 👍

hallelujah commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM centos/ruby-24-centos7
+FROM centos/ruby-25-centos7
```
I would use UBI instead of RHSCL CentOS

gsaslis replied:
Good point. 👍 

`registry.access.redhat.com/rhscl/ruby-25-rhel7:2.5` is UBI-based. let's go for that?

duduribeiro replied:
@gsaslis we tried but without success 😞 

Our system-builder needs a lot of xdg- packages to have chrome installed on it to be used on CI stuffs. UBI images don't have it and neither the package available to install. We can install it using centos packages but we can't guarantee to not make a mess in the packaging stuff of RHEL.

gsaslis replied:
@duduribeiro do you have that list of packages missing from UBI images handy anywhere?

duduribeiro replied:
@gsaslis AFAIK, these:

```
	libXss.so.1()(64bit) is needed by google-chrome-stable-63.0.3239.108-1.x86_64
	libatk-1.0.so.0()(64bit) is needed by google-chrome-stable-63.0.3239.108-1.x86_64
	libgconf-2.so.4()(64bit) is needed by google-chrome-stable-63.0.3239.108-1.x86_64
	libgdk-3.so.0()(64bit) is needed by google-chrome-stable-63.0.3239.108-1.x86_64
	libgdk_pixbuf-2.0.so.0()(64bit) is needed by google-chrome-stable-63.0.3239.108-1.x86_64
	libgtk-3.so.0()(64bit) is needed by google-chrome-stable-63.0.3239.108-1.x86_64
	xdg-utils is needed by google-chrome-stable-63.0.3239.108-1.x86_64
```

hallelujah committed:
[Dockerfile] Do not set RAILS_ENV and remove CMD

We should not set RAILS_ENV as it is persisted in all layers
Making it hard to use the system-builder for community image

Also remove CMD as the script/jenkins.sh does not exist anymore

hallelujah commented:
We can't delete this branch @duduribeiro I just reverted it
quay.io has a build relying on this branch
